Content.Client/_RMC14/EyeProtection/RMCEyeProtectionSystem.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Content.Shared._RMC14.EyeProtection;
+using Robust.Client.Graphics;
+using Robust.Client.Player;
+using Robust.Shared.Player;
+
+namespace Content.Client._RMC14.EyeProtection;
+
+public sealed class RMCEyeProtectionSystem : RMCSharedEyeProtectionSystem
+{
+    [Dependency] private readonly IOverlayManager _overlayManager = default!;
+    [Dependency] private readonly IPlayerManager _player = default!;
+
+    private RMCEyeProtectionOverlay _overlay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<RMCEyeProtectionComponent, ComponentInit>(OnEyeProtectionInit);
+        SubscribeLocalEvent<RMCEyeProtectionComponent, ComponentShutdown>(OnEyeProtectionShutdown);
+
+        SubscribeLocalEvent<RMCEyeProtectionComponent, LocalPlayerAttachedEvent>(OnEyeProtectionAttached);
+        SubscribeLocalEvent<RMCEyeProtectionComponent, LocalPlayerDetachedEvent>(OnEyeProtectionDetached);
+
+        _overlay = new();
+    }
+
+    private void OnEyeProtectionAttached(EntityUid uid, RMCEyeProtectionComponent component,  LocalPlayerAttachedEvent args)
+    {
+        _overlayManager.AddOverlay(_overlay);
+    }
+
+    private void OnEyeProtectionDetached(EntityUid uid, RMCEyeProtectionComponent component, LocalPlayerDetachedEvent args)
+    {
+        _overlayManager.RemoveOverlay(_overlay);
+    }
+
+    private void OnEyeProtectionInit(EntityUid uid, RMCEyeProtectionComponent component, ComponentInit args)
+    {
+        if (_player.LocalEntity == uid)
+            _overlayManager.AddOverlay(_overlay);
+    }
+
+    private void OnEyeProtectionShutdown(EntityUid uid, RMCEyeProtectionComponent component, ComponentShutdown args)
+    {
+        if (_player.LocalEntity == uid)
+        {
+            _overlayManager.RemoveOverlay(_overlay);
+        }
+    }
+}
```

Comment:
this could be massively reworked to use the ``ToggleClothingSystem``

here's an example in YML

```yml
- type: entity
  parent: [ClothingEyesBase, BaseToggleClothing]
  id: ClothingEyesVisorCamera
  name: camera visor
  description: Advanced glasses are designed to look like through cameras. Useful for working with Station AI.
  components:
  - type: Sprite
    sprite: Clothing/Eyes/Glasses/camera.rsi
    layers:
    - state: icon
      map: [ "enum.ToggleVisuals.Layer" ]
  - type: Clothing
    sprite: Clothing/Eyes/Glasses/camera.rsi
  - type: IdentityBlocker
    coverage: EYES
  - type: ToggleClothing
    action: ActionToggleVisorCamera
    disableOnUnequip: true
  - type: ComponentToggler
    parent: true
    components:
    - type: StationAiOverlay
  - type: Appearance
  - type: GenericVisualizer
    visuals:
      enum.ToggleVisuals.Toggled:
        enum.ToggleVisuals.Layer:
          True: {state: icon-on}
          False: {state: icon}
```

Reply:
Sorry for the long delay in responding, got distracted by other PRs. I'm currently in the process of reworking this PR because back when I first wrote this I really had no idea what I was doing.